src/packagedcode/rubygems.py

```diff
@@ -648,92 +629,172 @@ def normalize(gem_data, known_fields=known_fields):
-        self.summary = spec.get('summary')
-        self.author = spec.get('author')
-        self.authors = spec.get('authors')
-        # can be a list
-        self.email = spec.get('email')
-
-
-        self.spec['licenses'] = self.map_licenses()
-        self.make_unique()
-
-    def __str__(self):
-        return '<{}: {}>'.format(self.__class__.__name__, self.location)
-
-    def make_unique(self):
-        """
-        Ensure that lists in the spec only contain unique values.
-        """
-        new_spec = {}
-        for key, value in self.spec.items:
-            if isinstance(value, list):
-                newlist = []
-                for item in value:
-                    if item not in newlist:
-                        newlist.append(item)
-                new_spec[key] = newlist
-            else:
-                new_spec[key] = value
-        return new_spec
-
-    def get_description(self):
-        """
-        Using 'description' over 'summary' unless summary contains
-        more data.
-        See http://guides.rubygems.org/specification-reference/
-        Note that it is common to see this is spec files: s.description = s.summary
-        """
-        description = self.spec.get('description', '')
-        summary = self.spec.get('summary', '')
-
-        content = description
-        # FIXME: we should join these.
-        if len(summary) > len(description):
-            content = summary
-
-        content = ' '.join(content.split())
-        return content.strip()
-
-    def get_email(self):
-        """
-        Join the list of emails as a comma-separated string.
-        """
-        email = self.spec.get('email', u'')
-        if isinstance(email, list):
-            email = u', '.join(email)
-        return email
-
-    def map_licenses(self):
-        licenses = self.spec.get('licenses', [])
-        if not isinstance(licenses, list):
-            licenses = [licenses]
-
-        mapped_licenses = []
-        for lic in licenses:
-            mapped_license = LICENSES_MAPPING.get(lic, None)
-            if mapped_license:
-                mapped_licenses.append(mapped_license)
-            else:
-                if TRACE:
-                    logger.warning('WARNING: {}: no license mapping for: "{}"'.format(self.filename, lic))
-        return mapped_licenses
+    dependencies = gemspec_data.get('dependencies')
+    package_dependencies = []
+    if dependencies:
+        for dep_name, dep_version in dependencies.items():
+            package_dependencies.append(
+                models.DependentPackage(
+                    purl=PackageURL(
+                        type='gem', name=dep_name).to_string(),
+                    scope='dependencies',
+                    is_runtime=True,
+                    is_optional=False,
+                    requirement=dep_version,
+                )
+            )
+    package.dependencies=package_dependencies
+    return package
```

Comment:
You are returning the package here instead of Yielding (like your docstring states). Either convert this to a generator or modify your function's docstring.

Reply:
Done :)